### ISSUE TITLE
Clean up rc-cgroup.sh for clarity

### ISSUE THREAD

**Issue by tokiclover** [OpenRC/openrc#39]
Please, consider this patch in an attempt to clean up rc-cgroup.sh. I've just took a look the other day because I needed to make quick similar file for [Supervision-Scripts][1] and had a hard time going trough it because of poor coding style (somewhat confused at times) and bad variable naming scheme.

I suggest as well:
    * Renaming cgroup_cleanup() to cgroup_del_service() or cgroup_remove_service() to get more consistency on helpers naming (in par with cgroup_add_service()). This is not done in the current iteration because it would require patching runscripts.sh.
    * Use an environment variable--(e.g.) RC_CGROUP_NAME--for the use of 'openrc' group (name.)

Thanks.

(I am opening a bug request because I don't want to clone OpenRC and submit pull requests. Too much has to be done for that end. Moreover, upstream *can* or *could* completely ignore it--too much work and troubles for nothing. No thanks, Sir!)

NOTE: This is a copy of [bug][2] request.

[1]: https://github.com/tokiclover/supervision-scripts
[2]: https://bugs.gentoo.org/show_bug.cgi?id=541792

**Comment by qnikst** [comment]
I'm open for changes here. One bad thing is that 'croup_cleanup' is a part of public API, I.e. could be used by users, that have scripts that depends on it. So I'll try to check all stuff here and write my plan to improve things here, so you can say if you are OK with that.

**Comment by tokiclover** [comment]
No probleme. Another valid point on why it's only suggested and not done on the actual patch.

**Comment by tokiclover** [comment]
Here is a next [round][1] (v2) of the previous patch.

I was lucky that WilliamH noticed a usefull hint on the bug discussion and took advantage of it to introduce CG_{NAME,PATH,ROOT} to further clean up the file.

He also mentionned the trouble of patching at hand... but there is actually nothing much to do because there is actually only the addition of the description line. So, no bother to clone such a huge repo for an _almost_ tiny clean up. So, here comes v2!

NOTE: Expiration time is set to one month and not never to bpast. So, grab it while it's hot and fresh!

[1]:https://bpaste.net/show/6ffc5f063a52

**Comment by tokiclover** [comment]
Next round of the patch! [v3][1] comes rather early... I've just noticed that the last bits of `cgroup_cleanup()` are meaningless since the cleanup process is managed by the service (control) group, `croup_get_state()` would always report a (false-)negative state. Unless a special possibly costly check bits are added to verify that only the cleaner PID is the only remaining process, reporting the state is completly meaningless. So, removing the useless bits because I don't think adding that overhead worth the trouble.

Comments?

[1]:https://bpaste.net/show/dfe248c5e4bd

**Comment by tokiclover** [comment]
And [v4](https://bpaste.net/show/3bba85640abc) bringing in back a not-so-useless `ebegin` message in `cgroup_cleanup()`.

**Comment by williamh** [comment]
I am unable to apply this patch. The following command sequence is how I am attempting to apply it to master, and the "git apply" command fails:

cd /my/openrc/repo
wget -O  patch https://bpaste.net/raw/3bba85640abc
git apply patch

I suspect that the pastebin is corrupting the patch, and I have no idea how to fix that.

**Comment by tokiclover** [comment]
This is a copy/paste host, so copy & paste the file instead of downloading java & possibly html bits. It'd be interesting if it worked. There's nowhere to attach the file because the bug request is closed.

**Comment by williamh** [comment]
I'm not exactly sure what you mean wrt downloading java/html bits; there are none in the result if you run the wget command I gave above.

If by bug request you mean bugzilla bug, it was closed because it is a duplicate of this issue; there is no need to have both of them opened.

**Comment by tokiclover** [comment]
Of course you're always finding dubious $%*t to say in *any* case showing your *good* will to help a mere contributor. You *should* know that the patch wasn't generated with git because I said previously that I wasn't going to clone and format the patch for _your_ liking and hen _dump_ it whenever you feel like it because... after all, you're the supposedly benevolent _dictator_ here.

Now I don't care whatever may happen with your current liking and mood. Just drop everything and close this issue if you feel like it.

Leaving everything to you. Meaning, you can consider the patch completly broken.

Take care--of OpenRC project, that is. But I have more than doubts about it.

**Comment by williamh** [comment]
I can apply the patch manually if it comes to that, I just would rather apply it with "git apply", because that is far less error prone than applying it manually. So, I was just hoping you would give me an idea of why using wget to pull down the patch doesn't work.

I'm sorry you felt like I was attacking you, that wasn't the case at all.

**Comment by grknight** [comment]
@tokiclover Copy/pasting patches from a site is error prone due to mismatched spacing.  You may not get the tabs required instead of spaces or the site could interpret tabs as spaces.  The patch fails to apply that way.

**Comment by tokiclover** [comment]
Seriously, WilliamH, you don't have to. There's nothing of the sort, it's just, I was taking an option out early to avoid wasting unecessary time (from both sides.) You may have felt I was attacking you *personaly* instead, but certainly not the opposite, which is not my intent. Why should I have been, when you're just reiterating your preferences, again, with a certain disdain from what I have already said--at least twice? It just seemed, from where I stand, that it was going to happen... again with that feigned ignorence, without taking the time to look into the matter at hands instead--form before content. So, I've simply opted out.

If a punctal patch cannot get in, be it then.

**Comment by williamh** [comment]
@tokiclover:
I have not rejected your patch; I'm just trying to figure out the least error-prone way to apply it, that's all. :-)

**Comment by williamh** [comment]
@tokiclover:
ok, I'm making some headway. the patch does have to be manually adjusted in a couple of ways to be able to be applied.

**Comment by dwfreed** [comment]
The first problem is that the patch was using Windows newlines, and everybody trying to apply it has Unix newlines.  After fixing the newlines problem, the patch still fails to apply hunk 2 against current git master, which is quite large, so I can't figure out why it fails to apply.  Given the number and extensiveness of the changes, I would say that the best way to submit this contribution would be a pull request consisting of several commits of small changes, especially given that you've basically rewritten the entire file, and so would be git-blamed for everything, and several of the changes you've made are arguably controversial, so being able to skip some portions while merging the rest would be preferred, and is easiest with several small commits.  The repo isn't really that large that it takes forever to clone (try linux.git sometime for comparison).

**Comment by tokiclover** [comment]
Ok WilliamH, posting the [file][1] to the closed [bug][2]--actual file has UTF-8 charset.

Sorry, dwfreed, I won't follow your contorted arguments. If you have something substantial to say regarding a particular part... hell, say it! instead of talking about... well, that, how to say it? undefined and not named suggested issues.

So _you_ too are saying that a punctual contributor _shall_ make all the way in? Fine with me. Well, then provide a worthy issue tracker where I can attach files then. Sorry, Sir? Single way or the Highway?!

[1]:https://bugs.gentoo.org/attachment.cgi?id=399266
[2]:https://bugs.gentoo.org/show_bug.cgi?id=541792

**Comment by dwfreed** [comment]
> Sorry, dwfreed, I won't follow your contorted arguments. If you have something substantial to say regarding a particular part... hell, say it! instead of talking about... well, that, how to say it? undefined and not named suggested issues.

It's hard to do that when the only contribution lives as a patch on a pastebin service.  This is why I'm suggesting to make it a pull request, so we can use Github's existing interface for comments and review.  Manually staring at a patch and copy/pasting hunks to point out issues is not my idea of fun.

> So you too are saying that a punctual contributor shall make all the way in? Fine with me. Well, then provide a worthy issue tracker where I can attach files then. Sorry, Sir? Single way or the Highway?!

Ultimately, yes, that's the case.  As the people responsible for reviewing your contribution, we decide how we want to handle that.  If you don't like it, you know where the door is; don't let it hit you on the way out.

**Comment by tokiclover** [comment]
So, this is your substantial input? Funny one--by your (previous) standard of *fun*. Indeed, funny.

**Comment by dwfreed** [comment]
> So, this is your substantial input? Funny one--by your (previous) standard of fun. Indeed, funny.

Like I said, I can't give substantial input because your contribution lives on a pastebin service, and I'm not manually copy/pasting hunks to comment on them, and staring at patch files trying to figure out what they do gives me a headache.  If you want your contribution reviewed and possibly accepted, make it a pull request.

**Comment by tokiclover** [comment]
2015-03-20 0:49 GMT+01:00 Doug Freed <notifications@github.com>:

> Closed #39 <https://github.com/OpenRC/openrc/issues/39>.
>
Yes Lieutenant Chief, Sir! With such good *will*, the details do not matter
the least; you *could*, mind you, have, at least, found a "fun[ny]" thing
to say instead of... repeating yourself,--or the Chief mantra,--*ad
infinitum.*

**Comment by trofi** [comment]
To make a pull request you need to do such things:
```
$ git clone https://github.com/openrc/openrc
$ cd openrc

# hack/fix
$ git commit -s -m "A meaningful message"

# the magic part: dev-vcs/hub
$ hub fork
$ git push tokiclover master
$ hub pull-request
```
I personally find it easier than pastebining and manually opening bug report.
